pkg/controller/podautoscaler/horizontal.go

```diff
@@ -129,7 +130,16 @@ func (a *HorizontalController) computeReplicasForCPUUtilization(hpa *extensions.
 		targetUtilization = hpa.Spec.CPUUtilization.TargetPercentage
 	}
 	currentReplicas := scale.Status.Replicas
-	currentUtilization, timestamp, err := a.metricsClient.GetCPUUtilization(hpa.Namespace, scale.Status.Selector)
+
+	if scale.Status.Selector == nil {
+		return 0, nil, time.Time{}, fmt.Errorf("selector is required")
+	}
+	// TODO: Eliminate deserialization/reserialization of the selector.
+	selector, err := labels.Parse(*scale.Status.Selector)
```

Comment:
assuming we want this the same as ReplicaSet/Deployments, the internal api type will have `*unversioned.LabelSelector` as the convertable struct form, which you can get a `labels.LabelSelector` interface from using `unversioned.LabelSelectorAsSelector(...)`